src/traces/parcoords/parcoords.js

```diff
@@ -456,18 +456,20 @@ module.exports = function(root, svg, parcoordsLineLayers, styledData, layout, ca
 
     parcoordsLineLayer
         .each(function(d) {
-
             if(d.viewModel) {
-                if(d.lineLayer) d.lineLayer.update(d);
-                else d.lineLayer = lineLayerMaker(this, d);
+                if((!d.lineLayer) ||
+                    (callbacks)) { // recreate in case of having callbacks e.g. restyle. Should we test for callback to be a restyle?
+                    d.lineLayer = lineLayerMaker(this, d);
+                } else d.lineLayer.update(d);
```

Comment:
https://github.com/plotly/plotly.js/issues/3101#issuecomment-430694449 implies that in an earlier version this worked correctly using `update` even if there *are* callbacks, yes? It would be much preferred if we can get it to work that way again.

Reply:
I don't see an update call in the old version 1.35.2. It may used to create new lines by default which is not efficient.

Reply:
ha you're right, there isn't even an update method in `lineLayer` at that version! That was added by... uh, [me](https://github.com/plotly/plotly.js/blame/6de004240acfba04987ded259c3511d2526d5240/src/traces/parcoords/lines.js#L295) in https://github.com/plotly/plotly.js/pull/2415 😳